### PR TITLE
/user/anonymized - return user json without personal data

### PR DESCRIPTION
**Pull request by negue**
Ping @Alys did I miss a property?

@crookedneighbor didn't know what to test except if api call has no errors..^^

### CODE REVIEW

**Comment by Alys** [comment]
Definitely:
`profile`
`purchased.plan`
`contributor` (allows easy identification of many contributors)

Probably `items.special.nyeReceived`, `items.special.valentineReceived` (any other gift cards?) because they can contain the names of the people who sent cards

Perhaps `newMessages` and `invitations` because they can contain party names and private guild names.

`achievements.challenges` can contain the names of private challenges, which could contain users' names.

I would not remove all of `auth` because you probably want `auth.timestamps`. Removing `auth.local` and `auth.facebook` should be enough.

**Comment by negue** [comment]
newMessages has only `{GROUP_ID: Boolean}` there are no names ; added all others :D, how could I miss profile... :/

File: test/api.mocha.coffee [hunk 1]
```diff
@@ -259,6 +259,12 @@ describe "API", ->
               expect(body.err).to.be "Task not found."
               done()
 
+    describe "Anonymized User", ->
+      it "/api/v2/user/anonymized", (done) ->
+        request.get(baseURL + "/user/anonymized").set("Accept", "application/json").end (res) ->
+          expect(res.statusCode).to.be 200
```
Comment (crookedneighbor) [inline test/api.mocha.coffee]: This test says that the route returned a 200, which is great, but it should also inspect the res.body and make sure that no personal details are included.

**Comment by crookedneighbor** [comment]
Challenge tasks include info about the people participating in the challenge.

Also, the reason this can't be merged automatically is because I split up the api tests into separate files. If you do the following, this branch will be up to date with develop and the test will be moved to `test/api/anonymized.coffee`:

```
git fetch upstream negue-anonymizedUser
git reset --hard upstream/negue-anonymizedUser
git push --force
```

^ Only do that if you have not made any other changes to your branch.

File: test/api.mocha.coffee (hunk 1 above)
Comment (negue) [inline test/api.mocha.coffee]: any good idea how test something like that?

**Comment by Alys** [comment]
`newMessages` can contain guild names:

```
    "newMessages": {
        "b1865fe0-ae59-4e19-9cd4-a3b38976c9bc": {
            "value": true,
            "name": "The Name of the Guild"
        }
    },
```

**Comment by crookedneighbor** [comment]
@negue @Alys Is this good to merge now?

**Comment by negue** [comment]
@crookedneighbor Yep

Ping @Alys ?

**Comment by Alys** [comment]
I haven't looked into it apart from the comments I made. Let me know if you wanted me to test the code. But if you're happy with it now, I'm happy. :)

**Comment by crookedneighbor** [comment]
@negue I refined the test a little before merging: https://github.com/HabitRPG/habitrpg/commit/30ef56353438f47298d24a6cc132d9497324f1d9

**Pull request merged**